includes/functions/functions_traffic.php

```diff
@@ -29,11 +29,11 @@ function zen_get_ip_address() {
             $ip = $_SERVER['HTTP_FORWARDED_FOR'];
         } elseif (isset($_SERVER['HTTP_FORWARDED'])) {
             $ip = $_SERVER['HTTP_FORWARDED'];
-        } else {
+        } elseif (isset($_SERVER['REMOTE_ADDR'])) {
             $ip = $_SERVER['REMOTE_ADDR'];
         }
     }
-    if (trim($ip) == '') {
+    if (trim((string)$ip) == '') {
```

Comment:
> Further, why does this particular server not have any of the associated $_SERVER keys isn't that an issue?

The context of this change is scripts run via Cron, where some of the normal things are not set.

Reply:
> Does every variable in Zen Cart code need to be type cast at every possible use?
> 
excellent question.  i could not agree more...

Reply:
> > Further, why does this particular server not have any of the associated $_SERVER keys isn't that an issue?
> 
> The context of this change is scripts run via Cron, where some of the normal things are not set.

Then why not detect that condition and handle it appropriately? There *is* some detectable condition to identify that a cron job is running isn't there? This overall change obscures the identification of an errant server configuration.

Even if the change is to address a cron job, the provided/identified trouble data wouldn't necessitate casting.

Reply:
I'm pretty sure that `$_SERVER['REMOTE_ADDR']` doesn't exist when running via cron/command-line, thus the cast requirement.

Reply:
currency_cron is an admin side cron job.  This is a storefront side cron job.

Reply:
> currency_cron is an admin side cron job. This is a storefront side cron job.

I don't understand the importance of that statement. 

Create a file in the catalog root that is similar to currency_cron.php. Modify the "help" info and the feature to be executed. I don't see anything on the file that is otherwise inherently admin or catalog of any importance.